databricks-sdk-java/src/main/java/com/databricks/sdk/core/error/AbstractErrorMapper.java

```diff
@@ -0,0 +1,37 @@
+package com.databricks.sdk.core.error;
+
+import com.databricks.sdk.core.DatabricksError;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+abstract class AbstractErrorMapper {
+  @FunctionalInterface
+  protected interface Rule {
+    DatabricksError create(String message, List<ErrorDetail> details);
+  }
+
+  public DatabricksError apply(int code, ApiErrorBody errorBody) {
+    String message = errorBody.getMessage();
+    String errorCode = errorBody.getErrorCode();
+    List<ErrorDetail> details = errorBody.getErrorDetails();
+    if (statusCodeMapping.containsKey(code)) {
+      return statusCodeMapping.get(code).create(message, details);
```

Comment:
I think this could cause some confusing behavior for old clients when new error codes are introduced. The error code provided in the API response is not explicitly propagated to these exception types. I think you might need two "rule" interfaces: one for ErrorCode-based exceptions (with well-known error codes, so only need parameters for message and details) and one for StatusCode-based exceptions (with dynamic error-codes, so you need an extra parameter for the error code).